### PR TITLE
Browser Bundle Improvements

### PR DESCRIPTION
**Pull request by mistermoe**
Addresses #199

### CODE REVIEW

**Comment by mistermoe** [comment]
halved the bundle size by importing the 1 lodash function we're using instead of the entire lib
image

**Comment by mistermoe** [comment]
included `firefox108` and `safari16` as build targets for browser bundle.

**Comment by mistermoe** [comment]
configured `karma` to run tests in `WebkitHeadless` and `FirefoxHeadless`

**Comment by mistermoe** [comment]
switched node stdlib polyfill lib from [@esbuild-plugins/node-globals-polyfill](https://www.npmjs.com/package/@esbuild-plugins/node-globals-polyfill) to [node-stdlib-browser](https://github.com/niksy/node-stdlib-browser). 

`node-globals-polyfill` had us stuck on `esbuild` version `0.14.39`. esbuild is on `0.17.4` now with some relatively large improvements including support for `hermes` as a build target. this is particularly useful for us as `hermes` is on its way to becoming the defacto JS engine for react native. 

Still blocked on bumping all the way up to `0.17.4` as it's not supported by [karma-esbuild](https://github.com/marvinhagemeister/karma-esbuild) yet which we use to test our browser bundle in headless browsers. [karma-esbuild PR](https://github.com/marvinhagemeister/karma-esbuild/pull/55) to support `esbuild` `0.17` is up though and approved. just waiting to be merged.

additionally, `node-globals-polyfill` seems largely unmaintained. [relevant issue here](https://github.com/remorses/esbuild-plugins/issues/1)

File: package.json
Comment (shamilovtim) [inline package.json:3]: is the `main` entrypoint still on the table to add?

**Comment by mistermoe** [comment]
published `v0.0.18-unstable.2c05feb-2023.1.24-11-24-27` to npm so that we can test. need to test the following environments:
* cjs (node)
* esm (node)
* browser
* react-native

@thehenrytsai can you pull this branch down and run `npm run test:browser` to make sure it works on your end?

File: package.json
Comment (mistermoe) [reply package.json:3]: just added! need to test and make sure it works now

**Comment by mistermoe** [comment]
fixed an issue. new unstable version is `0.0.18-unstable.c04518c-2023.1.24-11-57-22`

**Comment by thehenrytsai** [comment]
> @thehenrytsai can you pull this branch down and run `npm run test:browser` to make sure it works on your end?
@mistermoe all tests passed on my end, with a nit that only the last batch of WebkitHeadless test results seemed to have printed out to console.

File: package.json
Comment (thehenrytsai) [inline package.json:10]: For my learning: what's the reason for bringing back building cjs? Is this something required by react native?

**Review by thehenrytsai — APPROVED** [review]

File: package.json
Comment (shamilovtim) [reply package.json:10]: @thehenrytsai related:

https://github.com/facebook/metro/issues/670

Comment (shamilovtim) [reply package.json:10]: RN **does** use ESM but for whatever reason just doesn't support the `exports` field yet

Comment (shamilovtim) [inline package.json:10]: just wanted to comment that i'm not sure we necessarily need a `cjs` bundle. the esm ones should work in theory, as RN uses esm. you could try pointing the `main` field to the esm, or if that's a non starte you could try adding the [react-native directive](https://github.com/facebook/metro/issues/670#issuecomment-1099926849) instead of `main`

File: src/store/message-store-level.ts
Comment (shamilovtim) [inline src/store/message-store-level.ts:6]: good call

Comment (shamilovtim) [reply src/store/message-store-level.ts:6]: we can add an eslint rule in the future that prohibits importing all of lodash:

```json
   "no-restricted-imports": [
      "error",
      {
        "name": "lodash",
        "message": "Dont import entire lodash. Use a module instead, for example: import _uniqBy from 'lodash/uniqBy';"
      },
      "error",
      {
        "name": "lodash-es",
        "message": "Dont import entire lodash. Use a module instead, for example: import _uniqBy from 'lodash/uniqBy';"
      }
    ]
```

File: package.json
Comment (mistermoe) [reply package.json:10]: @thehenrytsai i felt bad for including a `main` field that wasn't `cjs` because that is a widely accepted convention.

File: package.json
Comment (shamilovtim) [reply package.json:3]: I can confirm the `main` entrypoint works however wonder if you definitely want cjs given that react native should work with esm and via the `react-native` directive instead of `main`

File: package.json
Comment (mistermoe) [reply package.json:10]: nice find @shamilovtim! i'll def do that instead. @thehenrytsai am going to yank out the cjs before merging this

**Review by shamilovtim — APPROVED** [review]

File: package.json
Comment (mistermoe) [reply package.json:10]: actually going to leave the cjs as electron doesn't yet support esm. @shamilovtim any preference on whether you want the react native directive to point to esm or cjs? i imagine ejs but just wanted to be sure

Comment (shamilovtim) [reply package.json:10]: probably esm. no idea about ejs file types specifically

Comment (thehenrytsai) [reply package.json:10]: K, based on the parallel conversations, I take it that `cjs` is now needed not because of metro/react, but because of electron. Got it.

**Review by thehenrytsai — APPROVED** [review]
Seriously, what do we do without you! 🥇

**Review by shamilovtim — APPROVED** [review]

**Pull request merged**
